### PR TITLE
gh-94165: Prevent PY_CHECK_SQLITE_FUNC from accumulating linker options 

### LINKED ISSUE #94165: PY_CHECK_SQLITE_FUNC() adds -lsqlite3 multiple times
**Bug report**

The configure.ac macro ``PY_CHECK_SQLITE_FUNC`` keeps on adding ``-lsqlite3`` over and over again. This seems to cause some problems on Emscripten with a static ``libsqlite3.a`` file.

```
configure:12473: checking for sqlite3.h
configure:12473: result: yes
configure:12496: gcc -c   -I$(srcdir)/Modules/_sqlite  conftest.c >&5
configure:12496: $? = 0
configure:12500: checking for sqlite3_bind_double in -lsqlite3
configure:12525: gcc -o conftest   -I$(srcdir)/Modules/_sqlite  -lsqlite3   conftest.c -lsqlite3  -ldl  >&5
configure:12525: $? = 0
configure:12534: result: yes
configure:12551: checking for sqlite3_column_decltype in -lsqlite3
configure:12576: gcc -o conftest   -I$(srcdir)/Modules/_sqlite  -lsqlite3   conftest.c -lsqlite3  -lsqlite3 -ldl  >&5
configure:12576: $? = 0
configure:12585: result: yes
...
configure:12806: checking for sqlite3_result_double in -lsqlite3
configure:12831: gcc -o conftest   -I$(srcdir)/Modules/_sqlite  -lsqlite3   conftest.c -lsqlite3  -lsqlite3 -lsqlite3 -lsqlite3 -lsqlite3 -lsqlite3 -lsqlite3 -ldl  >&5
```

Erlend is going to look into the problem.

### PR DESCRIPTION
**Pull request by erlend-aasland**
Fixes gh-94165

### CODE REVIEW

**Comment by erlend-aasland** [comment]
Similar to the curses macro.

**Comment by erlend-aasland** [comment]
Darn, there are some Ubuntu issues.

File: configure.ac [hunk 1]
```diff
@@ -3740,16 +3754,16 @@ dnl hence CPPFLAGS instead of CFLAGS.
         PY_CHECK_SQLITE_FUNC([sqlite3_trace])
       ])
       PY_CHECK_SQLITE_FUNC([sqlite3_value_double])
-      AC_CHECK_LIB([sqlite3], [sqlite3_load_extension],
+      PY_CHECK_SQLITE_FUNC([sqlite3_load_extension],
         [have_sqlite3_load_extension=yes],
         [have_sqlite3_load_extension=no]
       )
-      AC_CHECK_LIB([sqlite3], [sqlite3_serialize], [
+      PY_CHECK_SQLITE_FUNC([sqlite3_serialize], [
```
Comment (tiran) [inline configure.ac:3761]: The third argument to ``PY_CHECK_SQLITE_FUNC([sqlite3_serialize], ...)``  is blank. Wouldn't this set ``have_supported_sqlite3=no`` if sqlite is missing the feature?

Comment (erlend-aasland) [reply configure.ac:3761]: 🤦🏻‍♂️

Comment (erlend-aasland) [reply configure.ac:3761]: I added a dummy value. Alternatively we could for example add a fourth argument (required=true|false) to PY_CHECK_SQLITE_FUNC.

Comment (erlend-aasland) [reply configure.ac:3761]: I'll have to revisit this later today.

**Pull request closed without merging**